### DISCUSSION TITLE
Restrict files list to show only specific paths from config file

### DISCUSSION THREAD

**Discussion by eyalk** [sevdokimov/log-viewer#8] [Q&A]
Hi,

Currently when opening http://localhost:8111/ on windows it display the entire directories under c:\users\user
Is there a way to restrict files list to show only specific paths from config file? For example list only c:\log .
I believe this is a security issue as an attacker will know how to map the entire c: folders structures.

Thanks,
Eyal

**Comment by sevdokimov** [comment]
Hi Eyalk,

Yes, you can specify the visible paths in `log-viewer-1.4.0/config.conf`. By default there is the following config:
```
"logs = [
  {
    path: "**/*.log"
  }
]
```
This means show all files with ".log" extension. Since .log file can be located in any directory, all directories are visible too.

Replace it this 
```
logs = [
  {
    path: "c:/log/**/*.log"
  }
]
```
Only subdirectories of `c:/log/*` will be visible.  If you specify `c:/log/*.log`, only .log files in `c:/log` will be accessible, no subritectory.

Feel free to ask any questions

**Comment by eyalk** [comment]
Thanks. It works!
Great project Sergey.

I have another question:
I have a log file without an extension but only the file with the .log
extension can be opened in the viewer.
for example I have the following logs file in the directory c:\log:
c:\log\somelogfile
c:\log\somelogfile2.log


Only the file  somelogfile2.log can be opened in the web, the file without
the .log extension have a question mark (?) icon and doesn't open when I
click on it. Is there a way to open it?

Thanks,
Eyal

**Comment by sevdokimov** [comment]
The File Chooser recognizes a file as a log only if the file name is like "*.log" , "*.txt" , "*.log.1" and "*.log.2020-12-01" , but you can open any file by passing the path as a parameter in URL: `http://localhost:8111/log?path=c:/log/somelogfile`

Don't forget to configure access to any file located in c:\log\ , by default only the files with log extension are available:
Replace 
```
logs = [
  {
    path: "c:/log/**/*.log"
  }
]
```
with 
```
logs = [
  {
    path: "c:/log/*"
  }
]
```


What is the name of your file? Can I recognize that the file is a log automatically?

**Comment by eyalk** [comment]
Thanks for the fast reply
Will do that
